### ISSUE TITLE
Use test-driven development to write new quest functions reflecting new quest data structures

### ISSUE THREAD

**Issue by ddelgado60** [uchicago-cs/chiventure#986]
Currently, we are waiting on a pull request for the new switch from a linked list data structure to a non-binary tree data structure to contain achievements. Once this is done, we have new data structures that allow for passive and active quests, new reward types, and a stats requirement before being able to start a quest.
In particular, we plan to remodel existing code to account for this, as well as writing new functions for the new data types. We will also write new tests for the new functions as well.

Assignees: Devon, Dillon, Gabi
Team: quests

**Comment by ddelgado60** [comment]
We are still waiting for a pull request for updated and new data structures from quests_structs, so we believe that this issue will be postponed to sprint 4. After writing up tests, we will implement functions, and think we could expand this issue to the whole group rather than just us two.

**Comment by ddelgado60** [comment]
The quests/implement-trees team will be sending a pull request in the next few days, so after that is merged into main, we can begin on this issue. We began this issue a bit too early, since we didn't know that content described here would be weighing on the trees implementation.

**Comment by ddelgado60** [comment]
We created a new branch off of quests/implement-trees called implement_new_quest_fcts  with git checkout -b implement_new_quest_fcts quests/implement-trees. We made a pull request a few days ago with similar new structures, but we think it will be a better idea to branch off of this so everything can be pushed to dev together with no merge conflicts or integration hell.

**Comment by ddelgado60** [comment]
The new functions, function headers, and tests have been implemented in the implement_new_quest_fcts branch, but it has not been tested yet. Once this has been done, the branch will be merged into quests/implement-trees branch, and we can begin merging the whole thing as one to dev branch and begin writing a monkeypatched chiventure.

**Comment by ddelgado60** [comment]
Finished implementing these functions and tests, and this is now in dev.

**Comment by lilyehsani** [comment]
Issue Score: ✔️++

Comments:
Great work overall! One comment: Make sure to select your team's identifier under "Labels".
